### PR TITLE
SAN 5677 Node demo unit test

### PR DESCRIPTION
**Pull request by damienrunnable**
Adding unit testing for the demo. Run by setting the cmd to npm test

### CODE REVIEW

**Comment by runnabot-brother** [comment]
Deployed  [node-starter-web/SAN-5677-Node](http://e3z5m3-node-starter-web-staging-runnabledemo.runnable.ninja). [View on Runnable](https://app.runnable-gamma.com/RunnableDemo/SAN-5677-Node-node-starter-web).
<sub>Related containers: [node-starter-db](https://app.runnable-gamma.com/RunnableDemo/e3z5m3--node-starter-db)*— From [Runnable](http://runnable.com)*</sub>

File: ManualDocker [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# For creating a docker file by hand in runnable, it translates to the special syntax
```
Comment (damienrunnable) [inline ManualDocker]: I didn't want to lose how a manually created docker file works.

Comment (Myztiq) [inline ManualDocker]: Perhaps the dockerfile should be called TestDockerfile

File: ManualDocker [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+FROM node:argon
+MAINTAINER Runnable, Inc.
+
+# Cache NPM Install
+COPY ["/node-starter/package.json","/app/"]
```
Comment (Myztiq) [inline ManualDocker]: we should change the syntax here to `COPY /node-starter/package.json /app` to get rid of quotes and array notation

File: package.json [hunk 3]
```diff
@@ -19,7 +19,8 @@
   "scripts": {
     "prestart": "npm run migrate",
     "start": "node index.js",
-    "migrate": "echo 'No migrations to run.'"
+    "migrate": "echo 'No migrations to run.'",
+    "test": "mocha test"
```
Comment (Myztiq) [inline package.json]: should we explicitly target mocha? `node_modules/.bin/mocha test`

File: test/unit/todos.unit.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const todos = require('../../todos');
+const sinon = require('sinon');
+const mongoose = require('mongoose');
+
+describe('todos', function () {
+  let Todo;
+
+  beforeEach(() => {
+    Todo = mongoose.model('Todo');
+    sinon.stub(Todo, "find").yields(null, []);
```
Comment (Myztiq) [inline test/unit/todos.unit.js]: " -> '

File: test/unit/todos.unit.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  });
+
+  it('all should return all', function() {
+    let response = {send : () => {}};
```
Comment (Myztiq) [inline test/unit/todos.unit.js]:
```javascript
let response = {
  send: sinon.stub()
}
```

File: test/unit/todos.unit.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    sinon.stub(response, "send");
+
+    todos.all({}, response);
+
+    sinon.assert.calledOnce(response.send);
```
Comment (Myztiq) [inline test/unit/todos.unit.js]: Do we expect this to be called with any arguments?

File: test/unit/todos.unit.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+describe('todos', function () {
+  let Todo;
+
+  beforeEach(() => {
+    Todo = mongoose.model('Todo');
+    sinon.stub(Todo, "find").yields(null, []);
+  });
+
+  it('all should return all', function() {
```
Comment (Myztiq) [inline test/unit/todos.unit.js]: all what?

File: ManualDocker (hunk 1 above)
Comment (damienrunnable) [inline ManualDocker]: Sounds good

File: ManualDocker (hunk 2 above)
Comment (damienrunnable) [inline ManualDocker]: I thought they were needed when manually entering the docker file?

File: ManualDocker (hunk 1 above)
Comment (thejsj) [inline ManualDocker]: Maybe `Dockerfile.test`

File: ManualDocker (hunk 2 above)
Comment (thejsj) [inline ManualDocker]: 👍

File: package.json (hunk 3 above)
Comment (damienrunnable) [inline package.json]: Not sure, is their any reason one is better than the other?

File: test/unit/todos.unit.js (hunk 4 above)
Comment (damienrunnable) [inline test/unit/todos.unit.js]: will fix

File: test/unit/todos.unit.js (hunk 5 above)
Comment (damienrunnable) [inline test/unit/todos.unit.js]: will change

File: test/unit/todos.unit.js (hunk 6 above)
Comment (damienrunnable) [inline test/unit/todos.unit.js]: I'll look again but I don't think so.

File: test/unit/todos.unit.js (hunk 7 above)
Comment (damienrunnable) [inline test/unit/todos.unit.js]: Ill increase the description

File: ManualDocker (hunk 2 above)
Comment (damienrunnable) [inline ManualDocker]: Nm, tested without and works fine. Will remove.

**Pull request merged**